### PR TITLE
Fix example; closes #3582

### LINKED ISSUE #3582: Typos in examples
Typos and errors in the example [here](https://github.com/ethereum/solidity/blob/develop/docs/frequently-asked-questions.rst#sometimes-when-i-try-to-change-the-length-of-an-array-with-ex-arraynamelength--7-i-get-a-compiler-error-value-must-be-an-lvalue-why):

```
int8[] memory memArr;        // Case 1
memArr.length++;             // illegal
int8[5] storageArr;          // Case 2
somearray.length++;          // legal
int8[5] storage storageArr2; // Explicit case 2
somearray2.length++;         // legal
```

* `somearray` should be `storageArr`
* `somearray2` should be `storageArr2`
* examples 2 & 3 are illegal as well since size is explicitly defined

### PR DESCRIPTION
**Pull request by aaroncolaco**
PS. formatting changes are due to the `.editorconfig` file. Not made by me

### CODE REVIEW

File: docs/frequently-asked-questions.rst [hunk 1]
```diff
@@ -542,12 +542,25 @@ contract level) with ``arrayname.length = <some new length>;``. If you get the
...
+
+    contract C {
+        int8[] dynamicStorageArray;
+        int8[5] fixedStorageArray;
+
+        function f() {
+            int8[] memory memArr;        // Case 1
+            memArr.length++;             // illegal
+
+            int8[5] storage memArr2 = fixedStorageArray;   // Case 2
```
Comment (chriseth) [inline docs/frequently-asked-questions.rst]: `memArr2` is not the best name here :)

File: docs/frequently-asked-questions.rst [hunk 2]
```diff
@@ -542,12 +542,25 @@ contract level) with ``arrayname.length = <some new length>;``. If you get the
...
 
-    int8[] memory memArr;        // Case 1
-    memArr.length++;             // illegal
-    int8[5] storageArr;          // Case 2
-    somearray.length++;          // legal
-    int8[5] storage storageArr2; // Explicit case 2
-    somearray2.length++;         // legal
+    pragma solidity ^0.4.18;
+
+    contract C {
```
Comment (chriseth) [inline docs/frequently-asked-questions.rst]: You have to add a special comment here that this snippet does not compile. As part of our tests, we automatically try to compile all source snippets in the documentation.

File: docs/frequently-asked-questions.rst (hunk 1 above)
Comment (aaroncolaco) [reply docs/frequently-asked-questions.rst]: Oops!

File: docs/frequently-asked-questions.rst [hunk 3]
```diff
@@ -542,12 +542,27 @@ contract level) with ``arrayname.length = <some new length>;``. If you get the
 
 ::
 
-    int8[] memory memArr;        // Case 1
-    memArr.length++;             // illegal
-    int8[5] storageArr;          // Case 2
-    somearray.length++;          // legal
-    int8[5] storage storageArr2; // Explicit case 2
-    somearray2.length++;         // legal
+    /// This contract will not compile due to multiple errors
```
Comment (chriseth) [inline docs/frequently-asked-questions.rst]: It has to be `// This will not compile`

**Comment by chriseth** [comment]
Could you please squash the commits?

**Comment by aaroncolaco** [comment]
@chriseth done

**Review by chriseth — APPROVED** [review]

**Comment by aaroncolaco** [comment]
Hi @chriseth, this hasn't been merged yet. Is there anything I need to do?

**Comment by axic** [comment]
We were waiting for the CI tasks to finish.

**Pull request merged**